drivers/esp8266/sock/udp.c

```diff
@@ -0,0 +1,173 @@
+void ESP8266_Callback_WifiConnected(ESP8266_t* ESP8266){}
+void ESP8266_Callback_WifiConnectFailed(ESP8266_t* ESP8266){}
+void ESP8266_Callback_WifiGotIP(ESP8266_t* ESP8266){}
+void ESP8266_Callback_WifiIPSet(ESP8266_t* ESP8266){}
+void ESP8266_Callback_DHCPTimeout(ESP8266_t* ESP8266){}
+void ESP8266_Callback_WifiDetected(ESP8266_t* ESP8266, ESP8266_APs_t* ESP8266_AP){}
+void ESP8266_Callback_ServerConnectionActive(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){}
+void ESP8266_Callback_ServerConnectionClosed(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){}
+void ESP8266_Callback_ServerConnectionDataReceived(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection, char* Buffer){}
+void ESP8266_Callback_ServerConnectionDataSent(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){}
+void ESP8266_Callback_ServerConnectionDataSentError(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){}
+void ESP8266_Callback_ClientConnectionConnected(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){
+    curr_conn = Connection;
+    msg_send(NULL, api_pid);
+}
+void ESP8266_Callback_ClientConnectionError(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){}
+void ESP8266_Callback_ClientConnectionTimeout(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){}
+void ESP8266_Callback_ClientConnectionClosed(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){}
+uint16_t ESP8266_Callback_ClientConnectionSendData(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection, char* Buffer, uint16_t max_buffer_size){
+    /* restrain size to avoid memory corruption */
+    if(send_len>max_buffer_size){
+        send_len = max_buffer_size;
+    }
+    memcpy(Buffer, send_data, send_len);
+    return send_len;
+}
+void ESP8266_Callback_ClientConnectionDataSent(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){
+    msg_send(NULL, api_pid);
+}
+void ESP8266_Callback_ClientConnectionDataSentError(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection){}
+void ESP8266_Callback_ClientConnectionDataReceived(ESP8266_t* ESP8266, ESP8266_Connection_t* Connection, char* Buffer){}
+void ESP8266_Callback_PingStarted(ESP8266_t* ESP8266, const char* address){}
+void ESP8266_Callback_PingFinished(ESP8266_t* ESP8266, ESP8266_Ping_t* Pinging){}
+void ESP8266_Callback_FirmwareUpdateStatus(ESP8266_t* ESP8266, ESP8266_FirmwareUpdate_t status){}
+void ESP8266_Callback_FirmwareUpdateSuccess(ESP8266_t* ESP8266){}
+void ESP8266_Callback_FirmwareUpdateError(ESP8266_t* ESP8266){}
+void ESP8266_Callback_ConnectedStationsDetected(ESP8266_t* ESP8266, ESP8266_ConnectedStations_t* Stations){}
+void ESP8266_Callback_SNTPOk(ESP8266_t* ESP8266, ESP8266_SNTP_t* SNTP){}
+void ESP8266_Callback_SNTPError(ESP8266_t* ESP8266){}
+
+/* static functions */
+static void* _update_loop(void* args){
+    ESP8266_t* esp8266_p = (ESP8266_t*) args;
+    uint32_t last_wakeup = xtimer_now();
+    while(1){
+        xtimer_periodic_wakeup(&last_wakeup, 1000);
```

Comment:
Jep. The parser/interface code has a ESP8266_TimeUpdate function which takes the time difference in ms as a parameter. 1 is smallest value possible. The parser need the function for determining timeouts.